src/components/ProteinStructureView.vue

```diff
@@ -158,50 +202,59 @@ export default {
-        if (!this.stage) {
-          this.stage = new NGL.Stage(this.$refs.container)
-          this.stage.signals.clicked.add((pickingProxy) => {
-            if (pickingProxy) {
-              const atom = pickingProxy.atom || pickingProxy.closestBondAtom
-              if (atom?.residueIndex != null) {
-                this.$emit('clickedResidue', {residueNumber: atom.residueIndex + 1})
-              }
-              // console.log(atom.qualifiedName())
-            }
-          })
-          this.stage.signals.hovered.add((pickingProxy) => {
-            if (pickingProxy) {
-              const atom = pickingProxy.atom || pickingProxy.closestBondAtom
-              if (atom?.residueIndex != null) {
-                this.$emit('hoveredOverResidue', {residueNumber: atom.residueIndex + 1})
-              }
-              // console.log(atom.qualifiedName())
-            }
-          })
-        }
-        // rcsb://1crn
-        this.stage.removeAllComponents()
-        this.stage.loadFile(`rcsb://${this.selectedPdb.id}`, /*{defaultRepresentation: true}*/).then((component) => {
-          this.mainComponent = component
-          component.addRepresentation('cartoon', {colorScheme: self.colorScheme})
-          //this.stage.autoView()
+      if (this.selectedAlphaFold) {
+        const viewerInstance = new PDBeMolstarPlugin()
+        const options = {
+          customData: {
+            url: `https://alphafold.ebi.ac.uk/files/AF-${this.selectedAlphaFold.id}-F1-model_v4.cif`,
+            format: 'cif',
+          },
+          /** This applies AlphaFold confidence score colouring theme for AlphaFold model */
+          // alphafoldView: true,
+          hideControls: true,
+          bgColor: { r: 255, g: 255, b: 255 },
+          // hideCanvasControls: [
+          //   'selection',
+          //   'animation',
+          //   'controlToggle',
+          //   'controlInfo',
+          // ],
+          // sequencePanel: true,
+          landscape: true,
+          highlightColor: '#ffffff',
+          selection: {
+            data: this.selectionDataWithSelectedColorBy,
+          },
+          selectInteraction: false,
+        };
+        const viewerContainer = document.getElementById('pdbe-molstar-viewer-container')
+        viewerInstance.render(viewerContainer, options)
+        viewerInstance.events.loadComplete.subscribe(() => {
+            viewerInstance.plugin.layout.context.canvas3d.camera.state.fog = 0
+            viewerInstance.plugin.layout.context.canvas3d.camera.state.clipFar = false
+            viewerInstance.visual.tooltips({data:this.residueTooltips.value})
         })
-      } else {
-        //
+
+        document.addEventListener('PDB.molstar.click', (e) => {
+          this.$emit('clickedResidue', e.eventData)
+        })
+        document.addEventListener('PDB.molstar.mouseover', (e) => {
+          this.$emit('hoveredOverResidue', e.eventData)
+        })
+        this.viewerInstance = viewerInstance
       }
```

Comment:
We should probably clean up these event listeners in an [unmounted](https://vuejs.org/api/options-lifecycle#unmounted) hook. It seems like if they aren't explicitly cleaned up they'll persist even after the component has been unmounted.

Reply:
Yes, they were persisting! Good catch. Removing them on `beforeUnmount` now.